packages/server-wallet/src/wallet/index.ts

```diff
@@ -81,23 +82,36 @@ export class Wallet implements WalletInterface {
 
       const {outgoing, channelResult} = await Store.signState(
         channelId,
-        {
-          ...channelConstants,
-          turnNum: 0,
-          isFinal: false,
-          appData,
-          outcome,
-        },
+        {...channelConstants, turnNum: 0, isFinal: false, appData, outcome},
         tx
       );
 
       return {outbox: outgoing.map(n => n.notice), channelResults: [channelResult]};
     });
   }
 
-  async joinChannel(_args: JoinChannelParams): Result {
-    throw 'Unimplemented';
+  async joinChannel({channelId}: JoinChannelParams): Result {
+    const {outbox} = await knex.transaction(
+      async (tx): Promise<{outbox: any}> => {
+        const channel = await Store.getChannel(channelId, tx);
+
+        if (!channel)
+          throw new JoinChannel.JoinChannelError(JoinChannel.Errors.channelNotFound, {
+            channelId,
+          });
+
+        const nextState = getOrThrow(JoinChannel.joinChannel({channelId}, channel));
```

Comment:
`getOrThrow ` 👀

Reply:
There is a big, outstanding question is: what is the wallet API?
- When should you return exactly one `ChannelResult`? When should you return an array of `ChannelResult[]`?
- What do you do when you have some successful results but some errors?

`getOrThrow` is a fill in until we answer questions like that. It is not meant to stay, and has an explicit note as such.